setup.py

```diff
@@ -1,47 +1,68 @@
 #!/usr/bin/env python
 
-from __future__ import print_function
+# One environment variable influence this script.
+#
+# GEOS_CONFIG: the path to a geos-config program that points to GEOS version,
+# headers, and libraries.
 
-import ctypes
-import ctypes.util
+import errno
+import glob
+import logging
+import os
+import platform
+import re
+import shutil
+import subprocess
+import sys
 try:
     # If possible, use setuptools
     from setuptools import setup
     from setuptools.extension import Extension
     from setuptools.command.build_ext import build_ext as distutils_build_ext
-    from setuptools.command.install import install
 except ImportError:
     from distutils.core import setup
     from distutils.extension import Extension
     from distutils.command.build_ext import build_ext as distutils_build_ext
-    from distutils.command.install import install
-from distutils.cmd import Command
 from distutils.errors import CCompilerError, DistutilsExecError, \
     DistutilsPlatformError
 from distutils.sysconfig import get_config_var
-import errno
-import glob
-import os
-import platform
-import re
-import shutil
-import subprocess
-import sys
 
+# Get geos_version from GEOS dynamic library
+# TODO: find a way to load a specific dynamic library
+from shapely.libgeos import geos_version_string, geos_version
```

Comment:
Unfortunately this import brings back the problem: `shapely.libgeos` still does the ctypes check for the shared libraries at module level. What's the point of using `geos-config` if this check is still made?